incrementals-publisher/index.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * This Azure Function is responsible for processing information related to an
+ * incrementals release and bouncing the artifacts into Artifactory
+ */
+
+const fs    = require('fs');
+const fetch = require('node-fetch');
+const os    = require('os');
+const path  = require('path');
+const util  = require('util');
+
+const github      = require('./lib/github');
+const pipeline    = require('./lib/pipeline');
+const permissions = require('./lib/permissions');
+
+const JENKINS_HOST     = process.env.JENKINS_HOST || 'https://ci.jenkins.io';
+const INCREMENTAL_URL  = process.env.INCREMENTAL_URL || 'https://repo.jenkins-ci.org/incrementals/'
+const ARTIFACTORY_KEY  = process.env.ARTIFACTORY_KEY || 'invalid-key';
+
+const TEMP_ARCHIVE_DIR = path.join(os.tmpdir(), 'incrementals-');
+const mktemp           = util.promisify(fs.mkdtemp);
+
+/*
+ * Small helper function to make failing a request more concise
+ */
+const failRequest = (context, body) => {
+  context.res = {
+    status: 500,
+    body: body || 'Unknown error'
+  };
+  context.done();
+};
+
+
+module.exports = async (context, data) => {
+  const buildUrl = data.body.build_url;
+  /* If we haven't received any valid data, just bail early
+   */
+  if ((!buildUrl) || (!buildUrl.match(JENKINS_HOST))) {
```

Comment:
Not sure what `match` means here. Is there a `startsWith` function? And the prefix URL should end with a `/` I think.

Reply:
There is, that's a good point. Will change